.github/workflows/stlab.yml

```diff
@@ -95,7 +132,15 @@ jobs:
           cmake --build ../build/
 
       - name: Test
+        if: ${{ !startsWith(matrix.config.compiler, 'emscripten') }}
         shell: bash
         run: |
           cd ../build/
           ctest
+
+      - name: Test (emscripten)
+        if: ${{ startsWith(matrix.config.compiler, 'emscripten') }}
+        shell: bash
+        run: |
+          find ../build/test -name "stlab.test.*.html" -exec \
+            emrun --browser_args="--headless" --browser firefox --kill_start --kill_exit {} \;
```

Comment:
Did you consider using ctest with the default node-based runner for these tests?

Reply:
Yes. I tried to get ctest to work with this, but I made little progress.

First, we'd need to change the output type from `.html` to the default `.js`. If we don't, the ctest node runner tries to parse the HTML as javascript, and dies.

If we make that change, and run ctest, we fail with the following:

```
Test project /Users/demarco/build/stlab-debug
    Start 1: stlab.test.channel
1/9 Test #1: stlab.test.channel ...............***Failed    0.56 sec
failed to asynchronously prepare wasm: CompileError: WebAssembly.instantiate(): unexpected section <Exception> @+37542
Aborted(CompileError: WebAssembly.instantiate(): unexpected section <Exception> @+37542)
Aborted(Assertion failed: native function `__trap` called before runtime initialization)
Aborted(Assertion failed: native function `__trap` called before runtime initialization)
Aborted(Assertion failed: native function `__trap` called before runtime initialization)
Aborted(Assertion failed: native function `__trap` called before runtime initialization)
```

That `__trap` error is repeated until you kill it with CTRL-C. 

To the best of my knowledge, neither ctest nor node are supported environments for running code transpiled with emscripten. The only "running" documentation is [here](https://emscripten.org/docs/compiling/Running-html-files-with-emrun.html#running-html-files-with-emrun), which suggests the use of emrun.

Reply:
Okay. In that case I think going with your own toolchain makes sense. You can set CMAKE_CROSSCOMPILING_EMULATOR to an emrun launch.

I don't think it is true that CTest isn't a supported environment from running code transpiled with emscripten. It is a general tool that can be used to run any cross-compiled code provided an appropriate emulator.

Node must be a supported environment for at least some emscripten since that is what they explicitly set in `Emscripten.cmake`.